lib/buy/onramper/onramper_buy_provider.dart

```diff
@@ -71,9 +73,14 @@ class OnRamperBuyProvider extends BuyProvider {
       if (response.statusCode == 200) {
         final Map<String, dynamic> data = jsonDecode(response.body) as Map<String, dynamic>;
         final List<dynamic> message = data['message'] as List<dynamic>;
-        return message
+
+        final allAvailablePaymentMethods = message
             .map((item) => PaymentMethod.fromOnramperJson(item as Map<String, dynamic>))
             .toList();
+
+        recommendedPaymentType = allAvailablePaymentMethods.first.paymentMethodType;
```

Comment:
if the recommended is the first avilable payment method, then what's customPaymentMethod? why is it returned separately?

Reply:
At first, I thought the first available payment method was the recommended one. It worked for most locations I tested, but it turned out not to work for all, so I had to add an extra API call to getDefaults to fetch the recommended method.